pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -414,16 +454,29 @@ contract ManagedPool is ManagedPoolSettings {
         // for the period during which they are an LP within the pool: otherwise an LP could shift their share of the
         // AUM fees onto the remaining LPs in the pool by exiting before they were paid.
         uint256 actualSupply = virtualSupply + _collectAumManagementFees(virtualSupply);
+        uint256[] memory normalizedWeights = _getNormalizedWeights(tokens);
 
         (bptAmountOut, amountsIn) = _doJoin(
             sender,
             balances,
-            _getNormalizedWeights(tokens),
+            normalizedWeights,
             scalingFactors,
             actualSupply,
             userData
         );
 
+        // Do not check circuit breakers on proportional joins, which do not change BPT prices.
+        if (userData.joinKind() != WeightedPoolUserData.JoinKind.ALL_TOKENS_IN_FOR_EXACT_BPT_OUT) {
```

Comment:
I'd not have this special case to reduce the number of edge cases we support

Reply:
Ok; my only reservation is that we're greatly increasing the gas cost and therefore "punishing" proportional joins. If a proportional join were much cheaper than an "exact tokens in" join (even a proportional one), it would incentivize people to join this way, which helps keep the pool well-behaved: especially for pools with higher token counts.